packages/shared/src/utils/os/os_detect.ts

```diff
@@ -46,7 +46,8 @@ export const isMobileOs = () =>
 
 export const isTabletOs =
     /ipad|android 3.0|xoom|sch-i800|playbook|tablet|kindle/i.test(navigator.userAgent.toLowerCase()) ||
-    (/android/i.test(navigator.userAgent.toLowerCase()) && !/mobile/i.test(navigator.userAgent.toLowerCase()));
+    (/android/i.test(navigator.userAgent.toLowerCase()) && !/mobile/i.test(navigator.userAgent.toLowerCase())) ||
+    (/MacIntel|Linux/.test(navigator.platform) && navigator.maxTouchPoints > 0);
```

Comment:
Note: function for detection TabletOS was taken from [here](https://github.com/binary-com/deriv-app/pull/15123/files#diff-d05e01aebb22cbc5a1ee9d4830bedce5ace5a036dc9d4a8422c5083d0c259147). Failing test cases were also fixed there.